File: core/sources/document.ts
import { IndexedSourceBase, type Metadata } from "../types/source";

/**
 * Source containing a single text document.
 */
export class DocumentSource extends IndexedSourceBase {
  readonly kind = "document";

  /**
   * @param id Unique source ID
   * @param metadata Generated metadata for the source
   * @param content Document text
   */
  constructor(
    id: string,
    metadata: Metadata | undefined,
    private readonly content: string,
  ) {
    super(id, metadata);
  }

  get isIterable(): false {
    return false;
  }

  /** Returns the document text. */
  getContent: () => string = () => this.content;

  buildIndexInput(): string {
    return this.content;
  }

  buildDeepJudgeInput(): string {
    return this.content;
  }
}

File: python/src/contextrie/sources/document.py
from __future__ import annotations

from ..types import IndexedSourceBase, Metadata


class DocumentSource(IndexedSourceBase):
    """Source containing a single text document."""

    kind = "document"

    def __init__(
        self,
        id: str,
        metadata: Metadata | None,
        content: str,
        path: str | None = None,
    ) -> None:
        super().__init__(id=id, metadata=metadata, path=path)
        self._content = content

    @property
    def is_iterable(self) -> bool:
        return False

    def get_content(self) -> str:
        return self._content

    async def build_index_input(self) -> str:
        return self._content

    async def build_deep_judge_input(self) -> str:
        return self._content
